sdk/servicebus/Microsoft.Azure.ServiceBus/src/Message.cs

```diff
@@ -498,6 +499,24 @@ public DateTime EnqueuedTimeUtc
                 internal set => this.enqueuedTimeUtc = value;
             }
 
+            /// <summary>Gets or sets the state of the message.</summary>
+            /// <value>The state of the message. </value>
+            /// <remarks>
+            ///    State of the message can be Active or Deferred or Scheduled. Deferred messages have Deferred state,
+            ///    scheduled messages have Scheduled state, all other messages have Active state.
+            ///    This property is read-only.
+            /// </remarks>
+            public MessageState State
```

Comment:
Since we are adding a new field, we need to do API approve changes, this is why check in tests are failing:  Microsoft.Azure.ServiceBus.UnitTests.API.ApiApprovals.ApproveAzureServiceBus 
Please follow this: https://microsoft.sharepoint.com/teams/ServiceBus/_layouts/OneNote.aspx?id=%2Fteams%2FServiceBus%2FShared%20Documents%2FService%20Bus%20Notebook&wd=target%281.Development%2FDev.one%7C6DE07DBD-1AB5-4766-A80D-4A763DA7424D%2FMessaging%20DotNetCore%7CCEB4B1BA-155B-4484-B86D-FD11AC0D9692%2F%29
onenote:https://microsoft.sharepoint.com/teams/ServiceBus/Shared%20Documents/Service%20Bus%20Notebook/1.Development/Dev.one#Messaging%20DotNetCore&section-id={6DE07DBD-1AB5-4766-A80D-4A763DA7424D}&page-id={CEB4B1BA-155B-4484-B86D-FD11AC0D9692}&end